sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc

```diff
@@ -0,0 +1,976 @@
+of the ordered range resulting from sorting `val` from all work-items in the group.
+Elements are compared by `operator<`
+using the `exec` group helper object.
+For multi-dimensional groups, the order of work-items in the group is
+determined by their linear id.
+
+_Complexity_: Let `N` be the work-group size. `O(N*log(N)*log(N))` comparisons.
+
+_Constraints_: Only available if `GroupHelper` was created with a work-group or a sub-group and
+some associated scratch space.
+
+5._Mandates_: `comp` must satisfy the requirements of `Compare` from the {cpp} standard.
+
+_Returns_: The value returned on work-item `i` is the value in position `i`
+of the ordered range resulting from sorting `val` from all work-items in the
+`g` group with respect to the binary comparison function object `comp`
+using the `exec` group helper object.
+For multi-dimensional groups, the order of work-items in the group is
+determined by their linear id.
+
+_Complexity_: Let `N` be the work-group or sub-group size. `O(N*log(N)*log(N))` comparisons.
+
+_Constraints_: Only available if `GroupHelper` was created with a work-group or a sub-group and
+some associated scratch space.
+
+6._Effects_: Equivalent to: `return sorter(g, val)`.
+
+_Constraints_: All functions are available only if `Sorter` is a SYCL Sorter and
+it provides `operator()(Group, T)` overload.
+
+=== Functions with fixed-size arrays
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+  template <typename GroupHelper, typename T, std::size_t ItemsPerWorkItem, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> values, Properties properties = {}); // (1)
+
+  template <typename GroupHelper, typename T, typename U, std::size_t ItemsPerWorkItem, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> keys, sycl::span<U, ItemsPerWorkItem> values, Properties properties = {}); // (2)
+
+  template <typename GroupHelper, typename T, std::size_t ItemsPerWorkItem, typename Compare, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> values, Compare comp, Properties properties = {}); // (3)
+
+  template <typename GroupHelper, typename T, typename U, std::size_t ItemsPerWorkItem, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(GroupHelper gh, sycl::span<T, ItemsPerWorkItem> keys, sycl::span<U, ItemsPerWorkItem> values, Compare comp, Properties properties = {}); // (4)
+
+  template<typename Group, typename T, std::size_t ItemsPerWorkItem, typename Sorter, typename Properties = sycl::ext::oneapi::experimental::properties<>>
+  void sort_over_group(Group g, sycl::span<T, ItemsPerWorkItem> values, Sorter sorter, Properties properties = {}); // (5)

```

Comment:
```suggestion
  template<typename Group, typename T, std::size_t ItemsPerWorkItem, typename Sorter, typename Properties>
  void sort_over_group(Group g, sycl::span<T, ItemsPerWorkItem> values, Sorter sorter, Properties properties = sycl::ext::oneapi::experimental::properties{}); // (5)
```

Don't you want to do something like this instead?

Reply:
* It seems like the `Properties` template parameter will be deduced from the type of the `properties` function parameter.  Therefore, there is no need to provide a default for the template parameter.

* I  changed the default value for the function parameter (`properties`) only because I thought the code you had before would be a syntax error.  Is it allowed to default to `{}`?

Reply:
> Is it allowed to default to `{}`?

Yes. Here is the example: https://godbolt.org/z/7d4PceGnP